### PR TITLE
Revisions for 0188-get-interior-data-resumption.md

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
Revisions for #554
Signed-off-by: Alexander <akutsan@luxoft.com>

Added information in the proposal that documentation should be updated with hash id management. 

Removed information that count of ignition cycles configured by ini file ( this value is hardcoded in SDL)

### CODE REVIEW

**Comment by LuxoftAKutsan** [comment]
@theresalech  please see the revisions PR for  #554

**Comment by yang1070** [comment]
@LuxoftAKutsan 
In the proposal, the following shall be mentioned
- The resultCode should be changed to a warning for double subscription, providing information if an app is already subscribed.

- Documentation has to be added to hashID in registerAppInterface in the RPC spec including the MOBILE_API.xml file in Core. 
This means MOBILE_API.xml shall be updated, with something like `interior vehicle data subscription will be included in app resumption since version 5.x`  in the description of a proper RPC about app resumption (I'm not sure registerAppInterface is the one),

**Comment by yang1070** [comment]
@LuxoftAKutsan and @theresalech 
I'm not sure about how the following point being handled
- Hash ID Resumption information should be added to the Best Practices guide.

`It will be determined upon implementation if the existing documentation about app resumption (https://smartdevicelink.com/en/docs/hmi/master/basiccommunication/onappregistered/) can be used as a Best Practice Guide, or if new documentation is needed to ensure clarity for developers.`

Does that mean once the proposal is approved and is implemented, we need to either update https://smartdevicelink.com/en/docs/hmi/master/basiccommunication/onappregistered/ or add something new in the `Best Practice Guide`?

**Comment by theresalech** [comment]
@yang1070 

>>>It will be determined upon implementation if the existing documentation about app resumption (https://smartdevicelink.com/en/docs/hmi/master/basiccommunication/onappregistered/) can be used as a Best Practice Guide, or if new documentation is needed to ensure clarity for developers.

This means that we will be adding information to the Best Practices Guide (and this should be noted in the proposal), but upon implementation, we'll need to determine exactly what the information should contain.  For example, will the information in the Best Practices Guide be a copy of https://smartdevicelink.com/en/docs/hmi/master/basiccommunication/onappregistered/ or will we need to write new documentation in the Best Practices Guide to make things more clear?

@LuxoftAKutsan @yang1070 Please advise when the additional revisions described in https://github.com/smartdevicelink/sdl_evolution/pull/573#issuecomment-409918035 are complete and I'll review the PR.  Thanks!

**Comment by LuxoftAKutsan** [comment]
@yang1070 

>> The resultCode should be changed to a warning for double subscription, providing information if an app is already subscribed.

The application cannot be already subscribed to anything during registration ( registration actually was not finished so any subscriptions could no be processed) 

>> Documentation has to be added to hashID in registerAppInterface in the RPC spec including the MOBILE_API.xml file in Core. This means MOBILE_API.xml shall be updated, with something like interior vehicle data subscription will be included in app resumption since version 5.x in the description of a proper RPC about app resumption (I'm not sure registerAppInterface is the one), 

Will be added in MOBILE_API, not sure that it should be in the proposal.

**Comment by yang1070** [comment]
@LuxoftAKutsan  @kmoiseienko @theresalech 

>> The resultCode should be changed to a warning for double subscription, providing information if an app is already subscribed.

> The application cannot be already subscribed to anything during registration ( registration actually was not finished so any subscriptions could no be processed)

Yes, I understand that during the registration or resumption, double subscription cannot happen. But the change required here is NOT about resumption, it is about regular app subscribes twice or more. Current behavior is that SDL returns `SUCCESS` for the 2nd subscribe request. The change is SDL returns a `WARNING` and proper `info` for 2nd subscribe request if the app has already subscribed the same module. It is a change of current SDL, therefore need to be included in the proposal.

**Comment by LuxoftAKutsan** [comment]
@theresalech @yang1070  proposal updated. Added information related to resultcode `WARNING` in case of a double subscription. Please review

File: proposals/0188-get-interior-data-resumption.md [hunk 1]
```diff
@@ -42,6 +46,10 @@ Reverting subscriptions means internally removing information about this subscri
 In the case that after reverting the subscription, there is no application subscribed to a certain module type, SDL should send `GetInteriorData(IsSubscribe=false)` to HMI and clear cache for this module type.
 Interior vehicle data resumption error handling should follow the same rules as regular vehicle data resumption error handling. 
 
+## SetInteriorVehicleData behaviour changes:
```
Comment (theresalech) [inline proposals/0188-get-interior-data-resumption.md]: Please fix spelling of "behavior"

File: proposals/0188-get-interior-data-resumption.md [hunk 2]
```diff
@@ -42,6 +46,10 @@ Reverting subscriptions means internally removing information about this subscri
 In the case that after reverting the subscription, there is no application subscribed to a certain module type, SDL should send `GetInteriorData(IsSubscribe=false)` to HMI and clear cache for this module type.
 Interior vehicle data resumption error handling should follow the same rules as regular vehicle data resumption error handling. 
 
+## SetInteriorVehicleData behaviour changes: 
+
+If mobile application send `SetInteriorVehicleData (subscribe=true, moduleType=MODULE1)`, but the application is already subscribed on `MODULE1` module type the resultCode should be `WARNING` because of double subscription and info should contain information if an app is already subscribed.
```
Comment (theresalech) [inline proposals/0188-get-interior-data-resumption.md]: Please update to "If a mobile application sends..."

**Comment by LuxoftAKutsan** [comment]
@theresalech  fixed.

**Pull request merged**